pkgs/build-support/php/build-composer-project.nix

```diff
@@ -66,13 +66,18 @@ let
         inherit composer composer-local-repo-plugin;
         inherit (finalAttrs) patches pname src vendorHash version;
 
+        php = phpDrv;
         composerLock = previousAttrs.composerLock or null;
         composerNoDev = previousAttrs.composerNoDev or true;
         composerNoPlugins = previousAttrs.composerNoPlugins or true;
         composerNoScripts = previousAttrs.composerNoScripts or true;
         composerStrictValidation = previousAttrs.composerStrictValidation or true;
       };
 
+      passthru = (previousAttrs.passthru or {}) // {
```

Comment:
Actually,  this will shadow `php` externally:

```
nix-repl> (stdenv.mkDerivation {name = "foo"; php = 1; passthru.php = 2;}).php
2

nix-repl> (stdenv.mkDerivation (finalAttrs: {name = "foo"; php = 1; passthru.php = 2; c = finalAttrs.php;})).c 
1
```

We should probably drop this and do `stdenvNoCC.mkDerivation ({ inherit php; } // args)` to serve as an initial value and replacing `phpDrv` with `finalAttrs.php` instead.

Reply:
Maybe something like the following (untested):

```patch
--- a/pkgs/build-support/php/build-composer-project.nix
+++ b/pkgs/build-support/php/build-composer-project.nix
@@ -4,8 +4,10 @@ let
   buildComposerProjectOverride = finalAttrs: previousAttrs:
 
     let
-      phpDrv = finalAttrs.php or php;
-      composer = finalAttrs.composer or phpDrv.packages.composer;
+      initialArgs = {
+        inherit php;
+      };
+      composer = finalAttrs.composer or finalAttrs.php.composer;
       composer-local-repo-plugin = callPackage ./pkgs/composer-local-repo-plugin.nix { };
     in
     {
@@ -18,12 +20,12 @@ let
       nativeBuildInputs = (previousAttrs.nativeBuildInputs or [ ]) ++ [
         composer
         composer-local-repo-plugin
-        phpDrv
-        phpDrv.composerHooks.composerInstallHook
+        finalAttrs.php
+        finalAttrs.php.composerHooks.composerInstallHook
       ];
 
       buildInputs = (previousAttrs.buildInputs or [ ]) ++ [
-        phpDrv
+        finalAttrs.php
       ];
 
       patches = previousAttrs.patches or [ ];
@@ -62,7 +64,7 @@ let
         runHook postInstallCheck
       '';
 
-      composerRepository = phpDrv.mkComposerRepository {
+      composerRepository = finalAttrs.php.mkComposerRepository {
         inherit composer composer-local-repo-plugin;
         inherit (finalAttrs) patches pname src vendorHash version;
 
@@ -82,4 +84,4 @@ let
       };
     };
 in
-args: (stdenvNoCC.mkDerivation args).overrideAttrs buildComposerProjectOverride
+args: (stdenvNoCC.mkDerivation (initialArgs // args)).overrideAttrs buildComposerProjectOverride
```

Reply:
`initialArgs` is not in the scope in `stdenvNoCC`, I moved it one level up... but it's not working anyway.